internal/test/cleanup/cleanup.go

```diff
@@ -189,3 +195,86 @@ func NutanixTestResources(clusterName, endpoint, port string, insecure, ignoreEr
 	}
 	return nil
 }
+
+// TinkerbellTestResources cleans up machines by powering them down.
+func TinkerbellTestResources(inventoryCSVFilePath string, ignoreErrors bool) error {
+	hardwarePool, err := api.NewHardwareMapFromFile(inventoryCSVFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create hardware map from inventory csv: %v", err)
+	}
+
+	logger.Info("Powering off hardware: %+v", hardwarePool)
+	return powerOffHardwarePool(hardwarePool, ignoreErrors)
+}
+
+func powerOffHardwarePool(hardware map[string]*hardware.Machine, ignoreErrors bool) error {
+	ctx, done := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
this gives bmclib 2 minutes to run operations on all machines. is that the intended behavior? There are many scenarios where one machine could take the whole time. I would recommend you create a separate 2 minute context per `powerOffHardware` call.

Reply:
No this was not intentional, it should have been in powerOffHardware. Thanks for the catch!